hash.go

```diff
@@ -138,8 +138,15 @@ func hashPasswordVersion1(password []byte, userName string, modulus []byte) (res
 }
 
 func hashPasswordVersion0(password []byte, userName string, modulus []byte) (res []byte, err error) {
-	var b64Hash []byte
+	b64Hash := make([]byte, 88) // 64 bytes in base64
 	prehashed := sha512.Sum512(password)
 	base64.StdEncoding.Encode(b64Hash, prehashed[:])
+	clear(prehashed[:])
 	return hashPasswordVersion1(b64Hash, userName, modulus)
```

Comment:
Don't you need to clear b64Hash after calling hashPasswordVersion1 ?

Reply:
Will do